### PR TITLE
[Cmake] libdvd find module refactor

### PR DESCRIPTION
**Pull request by fuzzard**
## Description
Split libdvd* find module into the 3 distinct lib find modules.
Build tools/depends platforms during core cmake project

## Motivation and context
Use macro/functions as with other cmake based find modules

## How has this been tested?
Built ios/macos/win64/winuwp-32

## What is the effect on users?
N/A

## Types of change

- [ ] **Bug fix** (non-breaking change which fixes an issue)
- [ ] **Clean up** (non-breaking change which removes non-working, unmaintained functionality)
- [x] **Improvement** (non-breaking change which improves existing functionality)
- [ ] **New feature** (non-breaking change which adds functionality)
- [ ] **Breaking change** (fix or feature that will cause existing functionality to change)
- [ ] **Cosmetic change** (non-breaking change that doesn't touch code)
- [ ] **None of the above** (please explain below)

## Checklist:

- [x] My code follows the **[Code Guidelines](https://github.com/xbmc/xbmc/blob/master/docs/CODE_GUIDELINES.md)** of this project 
- [ ] My change requires a change to the documentation, either Doxygen or wiki
- [ ] I have updated the documentation accordingly
- [x] I have read the **[Contributing](https://github.com/xbmc/xbmc/blob/master/docs/CONTRIBUTING.md)** document
- [ ] I have added tests to cover my change
- [ ] All new and existing tests passed

### CODE REVIEW

File: cmake/modules/FindLibDvdCSS.cmake [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                          "CFLAGS=${CMAKE_C_FLAGS}"
+                          "LDFLAGS=${CMAKE_EXE_LINKER_FLAGS} ${LIBDVDCSS_FLAGS}")
+    set(BUILD_COMMAND ${MAKE_EXECUTABLE})
+    set(INSTALL_COMMAND ${MAKE_EXECUTABLE} install)
+    set(BUILD_IN_SOURCE 1)
+  endif()
+
+  BUILD_DEP_TARGET()
+
+else()
```
Comment (wsnipex) [inline cmake/modules/FindLibDvdCSS.cmake:85]: this branch will be called whenever dvcss was _disabled_

File: cmake/modules/FindLibDvdNav.cmake [hunk 2]
```diff
@@ -0,0 +1,156 @@
...
+    set(INSTALL_COMMAND ${MAKE_EXECUTABLE} install)
+    set(BUILD_IN_SOURCE 1)
+  endif()
+
+  BUILD_DEP_TARGET()
+
+  if(TARGET LibDvdRead::LibDvdRead)
+    add_dependencies(libdvdnav LibDvdRead::LibDvdRead)
+  endif()
+else()
```
Comment (wsnipex) [inline cmake/modules/FindLibDvdNav.cmake:102]: this branch will be called whenever the dvdnav target already _exists_

File: cmake/modules/FindLibDvdRead.cmake [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+    set(BUILD_IN_SOURCE 1)
+  endif()
+
+  BUILD_DEP_TARGET()
+
+  if(TARGET LibDvdCSS::LibDvdCSS)
+    add_dependencies(libdvdread LibDvdCSS::LibDvdCSS)
+  endif()
+
+else()
```
Comment (wsnipex) [inline cmake/modules/FindLibDvdRead.cmake:109]: same here, the target exists already

**Comment by enen92** [comment]
I might need some help with this stuff at some point. Any idea about what is needed in cmake to remove the dll being built to the videoplayer folder and treat those libs like any other dependency (with internal builds for all platforms rather than Linux if the system lib is found)?
It would be great to have those changes somewhere to support development and testing of the VFS stuff

**Comment by fuzzard** [comment]
with this PR in place, essentially its just drop FindLibDvd.cmake. whats left in there is the whole dll wrapping stuff

File: cmake/modules/FindLibDvdCSS.cmake [hunk 4]
```diff
@@ -0,0 +1,119 @@
...
+  # We require this due to the odd nature of github URL's compared to our other tarball
+  # mirror system. If User sets LIBDVDCSS_URL or libdvdcss_URL, allow get_filename_component in SETUP_BUILD_VARS
+  if(LIBDVDCSS_URL OR ${MODULE_LC}_URL)
+    set(LIBDVDCSS_URL_PROVIDED TRUE)
+  endif()
+
+  SETUP_BUILD_VARS()
+
+  if(NOT LIBDVDCSS_URL_PROVIDED)
+    # override FFMPEG_URL due to tar naming when retrieved from github release for ffmpeg
```
Comment (enen92) [inline cmake/modules/FindLibDvdCSS.cmake:30]: comment looks like a copy/paste error from the findffmpeg module

File: cmake/modules/FindLibDvdNav.cmake [hunk 5]
```diff
@@ -0,0 +1,143 @@
...
+  # We require this due to the odd nature of github URL's compared to our other tarball
+  # mirror system. If User sets LIBDVDNAV_URL or libdvdcss_URL, allow get_filename_component in SETUP_BUILD_VARS
+  if(LIBDVDNAV_URL OR ${MODULE_LC}_URL)
+    set(LIBDVDNAV_URL_PROVIDED TRUE)
+  endif()
+
+  SETUP_BUILD_VARS()
+
+  if(NOT LIBDVDNAV_URL_PROVIDED)
+    # override FFMPEG_URL due to tar naming when retrieved from github release for ffmpeg
```
Comment (enen92) [inline cmake/modules/FindLibDvdNav.cmake:37]: comment looks like a copy/paste error from the findffmpeg module

File: cmake/modules/FindLibDvdRead.cmake [hunk 6]
```diff
@@ -0,0 +1,151 @@
...
+  # We require this due to the odd nature of github URL's compared to our other tarball
+  # mirror system. If User sets LIBDVDREAD_URL or libdvdcss_URL, allow get_filename_component in SETUP_BUILD_VARS
+  if(LIBDVDREAD_URL OR ${MODULE_LC}_URL)
+    set(LIBDVDREAD_URL_PROVIDED TRUE)
+  endif()
+
+  SETUP_BUILD_VARS()
+
+  if(NOT LIBDVDREAD_URL_PROVIDED)
+    # override FFMPEG_URL due to tar naming when retrieved from github release for ffmpeg
```
Comment (enen92) [inline cmake/modules/FindLibDvdRead.cmake:39]: comment looks like a copy/paste error from the findffmpeg module

**Review by wsnipex — APPROVED** [review]

**Comment by fuzzard** [comment]
Had some thoughts the windows stuff wasnt quite right. Turned out that was the case, haha. Thats fixed now, so i'll merge after this build completes

**Comment by mackal** [comment]
In the various files added, the ${MODULE_LC}_URL parts aren't actually allowing for either case to work like intended. Trying to build with `-Dlibdvdread_URL=...` fails since libdvdread_URL is never copied to LIBDVDREAD_URL (which is the variable CMake actually uses)

Building with `-DLIBDVDREAD_URL=...` actually appears to be working (haven't finished building, but it has gotten past where it was stuck)

https://github.com/gentoo/gentoo/blob/60b45e63c730b5eee5e61c31050b2716e9ddcb36/media-tv/kodi/kodi-9999.ebuild#L299 Gentoo passes with the libs lowercase, which worked fine before.

**Comment by fuzzard** [comment]
a "fix" Pr is done #21612 , however i would suggest gentoo changes to uppercase usage. I personally would prefer not to have to handle upper/lower case. cmake just doesnt do case insensitive things well, so its just annoying in an already very verbose language.
I understand thats whats been allowed for the libdvd libraries so far, so ill let @wsnipex give final approval on which way we go.

**Pull request merged**